R/uvhydrograph-data.R

```diff
@@ -225,17 +225,56 @@ parseLabelSpacing <- function(data, info) {
   
   if (names(data) %in% c("series_corr", "series_corr_ref", "series_corr_up", "series_corr2")){
     limits <- info[[grep("lims_UV", names(info))]]
-    y_positions <- rep(limits$ylim[2], length(data[[1]]$time))
-    differences <- as.numeric(diff(data[[1]]$time))
-    if(length(differences) > 0) {
-      for (i in seq_len(length(differences))) {
-        if(abs(differences[i]) < 86400) {y_positions[i+1] <- y_positions[i]-(0.04*info$lims_UV$ylim[2])}
-        i <- i + 1
-      }
-      spacingInfo <- list(y=y_positions, label=seq(length(data[[1]]$time)))
-    } else {
-      spacingInfo <- list(y=y_positions, label=seq(length(data[[1]]$time)))
-    }
+    #Number of hours to offset labels by for display
```

Comment:
Yeah, so this is the part that took me the longest to figure out and it is pretty hacky as is. These numbers are the "width" of various things in hours once they are rendered onto the plot. So with the current font and size settings in uvhydrograph-styles for the labels one digit ends up being 4 hours "wide" when it is rendered onto the plot. I need to know the final rendered size of the boxes in order to check for overlap and this was the only way I could figure out to do that without rendering the plot before adding the text (in order to be able to use the width function to get the width that the text would actually be). These values all end up being in in hours and they are used to space the labels and check for overlap. If there is a cleaner way to get the final rendered size of things that would certainly be an improvement here. The 0.065 on line 237 and the 0.475 on line 275 are also arbitrary numbers that I got after changing the sizes and then rendering the plot until I found sizes that fit well. If there was a way to convert between times and the actual plot scale then these could be significantly cleaned up as well.